tests/application_tests/test_method_converters.py

```diff
@@ -0,0 +1,394 @@
+        self.assertEqual(self.qsm_grad_ab[0]["CompositionA"], "80.0")
+        self.assertTrue("CompositionC" in self.qsm_grad_ab[0])
+        self.assertTrue("CompositionD" in self.qsm_grad_ab[0])
+        self.assertFalse("CompositionC" in self.bsm_grad1[0])
+        self.assertFalse("CompositionD" in self.bsm_grad1[0])
+
+        # Run function
+        new_gradient_table = change_gradient_table(
+            self.bsm_grad1,
+            self.qsm_grad_ab,
+        )
+
+        # Check output
+        for grad1, grad2 in zip(self.bsm_grad1, new_gradient_table):
+            self.assertEqual(grad1["CompositionA"], grad2["CompositionA"])
+            self.assertEqual(grad1["CompositionB"], grad2["CompositionB"])
+            self.assertTrue("CompositionC" in grad2)
+            self.assertTrue("CompositionD" in grad2)
+
+    def test_qsm_to_bsm(self) -> None:
+        """Test change_gradient_table function for QSM to BSM
+        Notes:
+        Transfers gradient ab to the format of bsm_grad1 and checks if the output is the
+        same as bsm_grad1.
+        """
+        # Check input
+        self.assertEqual(self.qsm_grad_ab[0]["CompositionA"], "80.0")
+        self.assertEqual(self.bsm_grad1[0]["CompositionA"], "90.0")
+        self.assertTrue("CompositionC" in self.qsm_grad_ab[0])
+        self.assertTrue("CompositionD" in self.qsm_grad_ab[0])
+        self.assertFalse("CompositionC" in self.bsm_grad1[0])
+        self.assertFalse("CompositionD" in self.bsm_grad1[0])
+
+        # Run function
+        new_gradient_table = change_gradient_table(
+            self.qsm_grad_ab,
+            self.bsm_grad1,
+        )
+
+        # Check output
+        for grad1, grad2 in zip(self.qsm_grad_ab, new_gradient_table):
+            self.assertEqual(grad1["CompositionA"], grad2["CompositionA"])
+            self.assertEqual(grad1["CompositionB"], grad2["CompositionB"])
+            self.assertFalse("CompositionC" in grad2)
+            self.assertFalse("CompositionD" in grad2)
+
+    def test_qsm_to_qsm(self) -> None:
+        """Test change_gradient_table function for QSM to QSM
+        Notes:
+        Transfers gradient cd to the format of qsm_grad_ab and checks if the output is
+        the same as qsm_grad_ab.
+        """
+        # Check input
+        self.assertEqual(self.qsm_grad_cd[0]["CompositionC"], "70.0")
+        self.assertEqual(self.qsm_grad_ab[0]["CompositionA"], "80.0")
+        self.assertTrue("CompositionC" in self.qsm_grad_cd[0])
+        self.assertTrue("CompositionD" in self.qsm_grad_cd[0])
+        self.assertTrue("CompositionC" in self.qsm_grad_ab[0])
+        self.assertTrue("CompositionD" in self.qsm_grad_ab[0])
+
+        # Run function
+        new_gradient_table = change_gradient_table(
+            self.qsm_grad_cd,
+            self.qsm_grad_ab,
+        )
+
+        # Check output
+        for grad1, grad2 in zip(self.qsm_grad_cd, new_gradient_table):
```

Comment:
Shouldn't we just check the whole gradient table?
```suggestion
        assertequal(self.qsm_grad_cd, new_gradient_table)
```

Ditto for any like-to-like substitution.

Reply:
Yeah no idea why i wasnt doing that...

Reply:
I guess because of the qsm and bsm comparison